File: sonarQuest-frontend/src/app/login/Token.ts
import * as moment from 'moment';

export class Token {

  constructor(private jwt: string, private expires_at: number) {
  }

  public hasExpired(): boolean {
    return moment().isBefore(this.expires_at);
  }

  public getJwt(): string {
    return this.jwt;
  }

}

File: sonarQuest-backend/src/main/java/com/viadee/sonarQuest/controllers/login/Token.java
package com.viadee.sonarQuest.controllers.login;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Token {

    private final String jwt;

    @JsonProperty("expires_at")
    private final long expiresAt;

    public Token(final String jwt, final long expiresAt) {
        super();
        this.jwt = jwt;
        this.expiresAt = expiresAt;
    }

    public String getJwt() {
        return jwt;
    }

    public long getExpiresAt() {
        return expiresAt;
    }

}
